src/filters/network.rs

```diff
@@ -837,93 +956,11 @@ impl NetworkFilter {
             vec![tokens]
         }
     }
+}
 
-    pub fn is_exception(&self) -> bool {
-        self.mask.contains(NetworkFilterMask::IS_EXCEPTION)
-    }
-
-    pub fn is_hostname_anchor(&self) -> bool {
-        self.mask.contains(NetworkFilterMask::IS_HOSTNAME_ANCHOR)
-    }
-
-    pub fn is_right_anchor(&self) -> bool {
-        self.mask.contains(NetworkFilterMask::IS_RIGHT_ANCHOR)
-    }
-
-    pub fn is_left_anchor(&self) -> bool {
-        self.mask.contains(NetworkFilterMask::IS_LEFT_ANCHOR)
-    }
-
-    fn match_case(&self) -> bool {
-        self.mask.contains(NetworkFilterMask::MATCH_CASE)
-    }
-
-    pub fn is_important(&self) -> bool {
-        self.mask.contains(NetworkFilterMask::IS_IMPORTANT)
-    }
-
-    pub fn is_redirect(&self) -> bool {
-        self.mask.contains(NetworkFilterMask::IS_REDIRECT)
-    }
-
-    pub fn is_removeparam(&self) -> bool {
-        self.mask.contains(NetworkFilterMask::IS_REMOVEPARAM)
-    }
-
-    pub fn also_block_redirect(&self) -> bool {
-        self.mask.contains(NetworkFilterMask::ALSO_BLOCK_REDIRECT)
-    }
-
-    pub fn is_badfilter(&self) -> bool {
-        self.mask.contains(NetworkFilterMask::BAD_FILTER)
-    }
-
-    pub fn is_generic_hide(&self) -> bool {
-        self.mask.contains(NetworkFilterMask::GENERIC_HIDE)
-    }
-
-    pub fn is_regex(&self) -> bool {
-        self.mask.contains(NetworkFilterMask::IS_REGEX)
-    }
-
-    pub fn is_complete_regex(&self) -> bool {
-        self.mask.contains(NetworkFilterMask::IS_COMPLETE_REGEX)
-    }
-
-    fn is_plain(&self) -> bool {
-        !self.is_regex()
-    }
-
-    pub fn is_csp(&self) -> bool {
-        self.mask.contains(NetworkFilterMask::IS_CSP)
-    }
-
-    fn third_party(&self) -> bool {
-        self.mask.contains(NetworkFilterMask::THIRD_PARTY)
-    }
-
-    fn first_party(&self) -> bool {
-        self.mask.contains(NetworkFilterMask::FIRST_PARTY)
-    }
-
-    fn for_http(&self) -> bool {
-        self.mask.contains(NetworkFilterMask::FROM_HTTP)
-    }
-
-    fn for_https(&self) -> bool {
-        self.mask.contains(NetworkFilterMask::FROM_HTTPS)
-    }
-
-    fn check_cpt_allowed(&self, cpt: &request::RequestType) -> bool {
-        match NetworkFilterMask::from(cpt) {
-            // TODO this is not ideal, but required to allow regexed exception rules without an
-            // explicit `$document` option to apply uBO-style.
-            // See also: https://github.com/uBlockOrigin/uBlock-issues/issues/1501
-            NetworkFilterMask::FROM_DOCUMENT => {
-                self.mask.contains(NetworkFilterMask::FROM_DOCUMENT) || self.is_exception()
-            }
-            mask => self.mask.contains(mask),
-        }
+impl NetworkFilterMaskHelper for NetworkFilter {
```

Comment:
do we need `impl NetworkFilterMaskHelper` for both `NetworkFilter` and `NetworkFilterMask`?

Reply:
Yes, to reduce the amount of changes in the codebase.